WeScan/Common/RectangleDetector.swift

```diff
@@ -17,18 +18,62 @@ struct RectangleDetector {
     /// - Parameters:
     ///   - image: The image to detect rectangles on.
     /// - Returns: The biggest detected rectangle on the image.
-    static func rectangle(forImage image: CIImage) -> CIRectangleFeature? {
-        let rectangleDetector = CIDetector(ofType: CIDetectorTypeRectangle, context: CIContext(options: nil), options: [CIDetectorAccuracy: CIDetectorAccuracyHigh])
-        
-        guard let rectangleFeatures = rectangleDetector?.features(in: image) as? [CIRectangleFeature] else {
-            return nil
+    static func rectangle(forImage image: CIImage, completion: @escaping ((Quadrilateral?) -> ())) {
```

Comment:
It would be great if we can use the `Result` enum here for completion, including the `Quadrilateral` if it succeeded and the occurred error if it failed

Reply:
It's quite late here, I'll look at implementing that tomorrow.